### PR TITLE
Add Spectron and Jest for integration tests

### PR DESCRIPTION
**Pull request by tsingh777**
RFC For testing framework. Added some basics test with hopes to expand. 
Relates to https://github.com/KhalisFoundation/sttm-desktop/issues/218

Run locally with `npm run test:integration`

Pending Task:
- [x] pass in the built app path so we can run tests against mac build and windows build
- [x] Integrate with CI (https://github.com/electron/spectron#on-travis-ci)
     - this might require building the app first ~without packaging~.
- [ ] Add windows app testing scripts
- [ ] Investigate how to include Testing DB so more scenarios can be tested.
- [ ] come up with meaningful scenarios to write tests for.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,7 +23,10 @@
     "build-css:main": "node-sass --output-style compressed --include-path www/src/scss www/src/scss/styles.scss www/assets/css/bundle.css",
     "build-css:obs": "node-sass --output-style compressed --include-path www/src/scss/obs www/src/scss/obs/styles.scss www/obs/bundle.css",
     "watch-css": "nodemon --watch www -e scss -x \"npm run build-css\"",
-    "postinstall": "electron-builder install-app-deps"
+    "postinstall": "electron-builder install-app-deps",
+    "test:pr": "npm run test && npm run pack:mac && npm run test:integration:mac",
```
Comment (tsingh777) [inline package.json]: Adds about 5mins to the build

File: tests/integration/app.test.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      const mainTabString = await app.client.getAttribute('#history-tab-content', 'class');
+      const mainTab = mainTabString.split(' ');
+      expect(mainTab).toContain('active');
+
+      await app.client.click('#themes-tab');
+      let classListString = await app.client.getAttribute('#themes-tab-content', 'class');
+      let classList = classListString.split(' ');
+      expect(classList).toContain('active');
+
+      await app.client.click('#insert-tab');
```
Comment (inderpreetsingh) [inline tests/integration/app.test.js]: Do you think these selectors might change when we refactor?

File: tests/integration/app.test.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  afterEach(() => {
+    if (app && app.isRunning()) {
+      return app.stop();
+    }
+    return false;
+  });
+
+  test('should test true is true', async () => { // true
+    expect(true).toBe(true);
+  });
```
Comment (inderpreetsingh) [inline tests/integration/app.test.js]: Possibly noob question, What would be the purpose of this?

File: tests/integration/app.test.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+      expect(classList).toContain('active');
+    });
+  });
+
+  describe('Themes', () => {
+    test('app contains default theme', async () => {
+      const classListString = await app.client.getAttribute('body', 'class');
+      const classList = classListString.split(' ');
+
+      expect(classList).toContain('light-theme');
```
Comment (inderpreetsingh) [inline tests/integration/app.test.js]: should default theme come from a common constant variable?

File: tests/integration/app.test.js (hunk 3 above)
Comment (tsingh777) [reply tests/integration/app.test.js]: was there to test that the test were running properly. can remove it.

File: tests/integration/app.test.js (hunk 4 above)
Comment (tsingh777) [reply tests/integration/app.test.js]: I can make it read from the `defaults.json`. will make that change

File: tests/integration/app.test.js (hunk 2 above)
Comment (tsingh777) [reply tests/integration/app.test.js]: if they change, the test can be updated

**Comment by tsingh777** [comment]
Really out of date now. Will revisit after refactor.

**Pull request closed without merging**